### PR TITLE
chore: Split up network options and transaction options

### PR DESCRIPTION
**Pull request by niklasnatter**

### CODE REVIEW

File: packages/shared/src/stateChainGateway/__tests__/__snapshots__/index.test.ts.snap [hunk 1]
```diff
@@ -4,13 +4,7 @@ exports[`executeRedemption executes the redemption 1`] = `
 [
   "0x1234",
   {
-    "network": "sisyphos",
     "nonce": 1,
-    "signer": VoidSigner {
```
Comment (niklasnatter) [inline packages/shared/src/stateChainGateway/__tests__/__snapshots__/index.test.ts.snap:9]: this was the initial reason for the pr - we pass the `signer` and `network` to `ethers.js` at the moment. this causes ethers to throw an error that it does not expect a `signer` property

File: packages/shared/src/vault/executeSwap.ts [hunk 2]
```diff
@@ -106,43 +101,38 @@ const callNative = async (
...
 const callToken = async (
   params: TokenCallParams,
-  {
-    signer,
-    network,
-    vaultContractAddress: vaultAddress,
-    srcTokenContractAddress: tokenAddress,
-    ...opts
-  }: ExecuteOptions,
+  opts: SwapNetworkOptions,
```
Comment (zoheb391) [inline packages/shared/src/vault/executeSwap.ts:112]: i think i would rename opts to `networkOpts` everywhere we have two types of opts for the sake of clarity

**Review by zoheb391 — APPROVED** [review]
the documentation changes should go to the new doc repo. (push to main) instead of the old one.

File: packages/shared/src/vault/executeSwap.ts (hunk 2 above)
Comment (niklasnatter) [reply packages/shared/src/vault/executeSwap.ts:112]: sounds good!

**Pull request merged**